web/src/components/DatePicker.tsx

```diff
@@ -0,0 +1,113 @@
+import { DateTime } from 'luxon'
+import React, { ReactElement, useEffect, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import styled from 'styled-components'
+
+import { CalendarTodayIcon } from '../assets'
+import dimensions from '../constants/dimensions'
+
+const DateContainer = styled.div`
+  width: fit-content;
+  position: relative;
+`
+
+const StyledInput = styled.input`
+  min-width: 316px;
+  height: 56px;
+  padding: 0 16px;
+  border-radius: 8px;
+  border-color: ${props => props.theme.colors.themeColorLight};
+  border-width: 3px;
+  border-style: solid;
+
+  &&& {
+    &:focus {
+      outline: none;
+      border-color: ${props => props.theme.colors.themeColor};
+    }
+
+    &::-webkit-calendar-picker-indicator {
+      background: url(${CalendarTodayIcon}) no-repeat center center;
+      width: 40px;
+      height: 40px;
+      border-radius: 50%;
+      cursor: pointer;
+      background-color: ${props => props.theme.colors.textDisabledColor};
+    }
+
+    &:focus::-webkit-calendar-picker-indicator {
+      background-color: ${props => props.theme.colors.themeColorLight};
+    }
+  }
+
+  @media ${dimensions.mediumViewport} {
+    min-width: 240px;
+  }
+`
+
+const StyledTitle = styled.span`
+  background-color: ${props => props.theme.colors.backgroundColor};
+  position: absolute;
+  top: -12px;
+  left: ${props => (props.theme.contentDirection === 'rtl' ? 'auto' : '12px')};
+  right: ${props => (props.theme.contentDirection === 'rtl' ? '12px' : 'auto')};
+  padding: 2px 5px;
+  font-size: 12px;
+  font-weight: 400;
+`
+
+const StyledError = styled.div`
+  font-size: 12px;
+  font-weight: bold;
+  color: ${props => props.theme.colors.invalidInput};
+`
+
+export type DatePickerProps = {
+  title: string
+  date?: DateTime | null
+  setDate: (date: DateTime | null) => void
+  error?: string
+}
+
+const isValidIsoDate = (date: string): boolean => {
+  try {
+    DateTime.fromISO(date)
+    return true
+  } catch (_) {
+    return false
+  }
+}
+
+const DatePicker = ({ title, date, setDate, error }: DatePickerProps): ReactElement => {
+  const { t } = useTranslation('events')
+  const [tempDate, setTempDate] = useState(date?.toISODate() ?? '')
+  const isInvalidDate = tempDate !== '' && isValidIsoDate(tempDate) === false
+  const shownError = error || (isInvalidDate ? t('invalidDate') : undefined)
+  useEffect(() => {
+    setTempDate(date?.toISODate() ?? '')
+  }, [date])
+
+  const handleDateChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTempDate(event.target.value)
+    try {
+      setDate(DateTime.fromISO(event.target.value))
+    } catch (_) {
+      // Invalid date format, do not update the state
```

Comment:
I like this much better than the way you implemented it in the native part where you commented out a function.

I still disagree that we should just ignore it, though, my reasoning is in the native part :)

Reply:
❌ We need to handle this error.
<img width="312" alt="image" src="https://github.com/user-attachments/assets/24aeb3b4-ddc8-446e-9bd6-d4c2b85686fb">

Reply:
I tried to do it like I did in native but for some reason the `<input type=date` is not letting the onChange event to do anything if there is out of range error...